ios/Classes/FlutterBleLibPlugin.m

```diff
@@ -205,25 +211,55 @@ - (void)readCharacteristicForDevice:(FlutterMethodCall *)call result:(FlutterRes
-                                  resolve:[self resolveForReadCharacteristic:result]
+                                  resolve:[self resolveForReadWriteCharacteristic:result]
                                    reject:[self rejectForFlutterResult:result]];
 }
 
 - (void)readCharacteristicForService:(FlutterMethodCall *)call result:(FlutterResult)result {
     [_manager readCharacteristicForService:[call.arguments[ARGUMENT_KEY_SERVICE_ID] doubleValue]
                         characteristicUUID:call.arguments[ARGUMENT_KEY_CHARACTERISTIC_UUID]
                              transactionId:[ArgumentValidator validStringOrNil:call.arguments[ARGUMENT_KEY_TRANSACTION_ID]]
-                                   resolve:[self resolveForReadCharacteristic:result]
+                                   resolve:[self resolveForReadWriteCharacteristic:result]
                                     reject:[self rejectForFlutterResult:result]];
 }
 
 - (void)readCharacteristic:(FlutterMethodCall *)call result:(FlutterResult)result {
     [_manager readCharacteristic:[call.arguments[ARGUMENT_KEY_CHARACTERISTIC_IDENTIFIER] doubleValue]
                    transactionId:[ArgumentValidator validStringOrNil:call.arguments[ARGUMENT_KEY_TRANSACTION_ID]]
-                         resolve:[self resolveForReadCharacteristic:result]
+                         resolve:[self resolveForReadWriteCharacteristic:result]
                           reject:[self rejectForFlutterResult:result]];
 }
 
+- (void)writeCharacteristicForDevice:(FlutterMethodCall *)call result:(FlutterResult)result {
+    [_manager writeCharacteristicForDevice:call.arguments[ARGUMENT_KEY_DEVICE_IDENTIFIER]
+                               serviceUUID:call.arguments[ARGUMENT_KEY_SERVICE_UUID]
+                        characteristicUUID:call.arguments[ARGUMENT_KEY_CHARACTERISTIC_UUID]
+                               valueBase64:[self base64encodedStringFromBytes:call.arguments[ARGUMENT_KEY_BYTES]]
+                                  response:(BOOL)call.arguments[ARGUMENT_KEY_WITH_RESPONSE]
+                             transactionId:[ArgumentValidator validStringOrNil:call.arguments[ARGUMENT_KEY_TRANSACTION_ID]]
+                                   resolve:[self resolveForReadWriteCharacteristic:result]
+                                    reject:[self rejectForFlutterResult:result]];
+}
+
+- (void)writeCharacteristicForService:(FlutterMethodCall *)call result:(FlutterResult)result {
+    [_manager writeCharacteristicForService:[call.arguments[ARGUMENT_KEY_SERVICE_ID] doubleValue]
+                         characteristicUUID:call.arguments[ARGUMENT_KEY_CHARACTERISTIC_UUID]
+                                valueBase64:[self base64encodedStringFromBytes:call.arguments[ARGUMENT_KEY_BYTES]]
+                                   response:(BOOL)call.arguments[ARGUMENT_KEY_WITH_RESPONSE]
+                              transactionId:[ArgumentValidator validStringOrNil:call.arguments[ARGUMENT_KEY_TRANSACTION_ID]]
```

Comment:
We're not validating the value of this argument, this is only used to cast from something to String. Perhaps it should be renamed, both the call and the class?

Reply:
Create an issue and merge.

Reply:
Issue created #181